src/views/TrackPositionAnimation/TrackPositionAnimationView.tsx

```diff
@@ -164,6 +167,13 @@ const TrackPositionAnimationView: FunctionComponent<TrackPositionAnimationProps>
         })
     }, [dataFrames])
 
+    // TODO: Full support of dynamic updates to frame set will require looking at AnimationState.frameData instead of dataFrames!
+
+    const { focusTime, setTimeFocus } = useTimeFocus()  // state imported from recording context
+    const findNearestTime = useTimeLookupFn(animationState)
+    useEffect(() => matchFrameToFocus(focusTime, findNearestTime, animationState, animationStateDispatch), [focusTime])
+    useEffect(() => matchFocusToFrame(animationState, focusTime, setTimeFocus), [animationState.isPlaying])
+
```

Comment:
Here's the time update code in action. The nearest-time-lookup function is provided by a custom hook that depends on the animation state's frame data (and thus should update correctly if, in the future, we have a dynamically loaded time series). The two `useEffect` hooks should control the triggers and direction of the time updates: match the frame to the focus if the focus changes; and match the focus to the frame if we change playback state.